packages/slice-machine/src/domain/customType.ts

```diff
@@ -425,6 +453,64 @@ export function reorderField(args: ReorderFieldArgs): CustomType {
   return newCustomType;
 }
 
+// if the UID is not existing in any section, it should be added to the main section
+export function addUIDField(label: string, customType: CustomType): CustomType {
+  const sectionId = getMainSectionId(customType);
+  if (typeof sectionId === "undefined") {
+    return customType;
+  }
+  const newFieldId = "uid";
+  const newField: UID = {
+    config: {
+      label,
+    },
+    type: "UID",
+  };
+
+  return addField({ customType, sectionId, newField, newFieldId });
+}
+
+export function updateUIDField(
+  label: string,
+  customType: CustomType,
+): CustomType {
+  const [sectionId, section] = getSectionWithUIDField(customType) ?? [];
+
+  const fieldId = Object.keys(section ?? {})?.find(
+    (key) => section?.[key]?.type === "UID",
+  );
+
+  const field = getUIDField(customType);
+
+  if (
+    !field ||
+    typeof fieldId === "undefined" ||
```

Comment:
💡 #idea

Why `typeof fieldId === "undefined"` and not as for field `!field`?

Reply:
typescript didn't like the other option: `Unexpected nullable string value in conditional`

Reply:
<img width="218" alt="Screenshot 2024-08-08 at 12 16 32" src="https://github.com/user-attachments/assets/dccd02e2-bc47-4990-bea4-92b9a3614903">

Hum I don't have that 🧐